stock_reserve_area/models/stock_quant.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2023 ForgeFlow SL.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    reserve_area_id = fields.Many2one(
+        "stock.reserve.area", compute="_compute_reserve_area_id", store=True
+    )
+
+    @api.depends("location_id")
+    def _compute_reserve_area_id(self):
+        for quant in self:
+            reserve_areas = self.env["stock.reserve.area"].search([])
+            for reserve_area in reserve_areas:
+                if reserve_area.is_location_in_area(quant.location_id):
+                    quant.reserve_area_id = reserve_area
+                    break
+
+    def _get_reserve_area_available_quantity(self, product_id, reserve_area_id):
+        self = self.sudo()
+        quants = self._gather_reserve_area(product_id, reserve_area_id)
+        rounding = product_id.uom_id.rounding
+        # for the moment not taking into account lots
+        # if product_id.tracking == 'none':
+        area_reserved_quantity = (
+            self.env["stock.move"]
+            .search(
+                [
+                    ("reserve_area_id", "=", reserve_area_id.id),
+                    ("product_id", "=", product_id.id),
+                ]
+            )
+            .mapped("area_reserved_availability")
+        )
+        available_quantity = sum(quants.mapped("quantity")) - sum(
+            area_reserved_quantity
+        )
+        return (
+            available_quantity
+            if float_compare(available_quantity, 0.0, precision_rounding=rounding)
+            >= 0.0
+            else 0.0
+        )
+
+    def _gather_reserve_area(self, product_id, reserve_area_id):
+        self.env["stock.quant"].flush(["reserve_area_id", "available_quantity"])
+        quants = self
+        for location_id in reserve_area_id.location_ids:
+            if reserve_area_id.is_location_in_area(location_id):
+                quants |= self._gather(product_id, location_id)
```

Comment:
This will likely be slow. Collect the ids and recreate the recordset by calling browse(ids) at the end

Reply:
@jbaudoux thanks for you suggestions. In this one, I don't understand what you mean. Quants are returned by the _gather method that takes the location as an argument. You mean doing 
`quants += self._gather(product_id, location_id).ids` ?

Reply:
yes
```
quant_ids += self._gather(product_id, location_id).ids
...
quants = self.browse(quant_ids)
```